src/service/navigation.js

```diff
@@ -489,3 +528,29 @@ function maybeExpandUrlParams(ampdoc, e) {
     target.setAttribute('href', newHref);
   }
 }
+
+/**
+ * A transformation applied to an anchor element, executed according to its
+ * priority.
+ */
+class AnchorMutator {
+  /**
+   * @param {function(!Element):!Element} callback
+   * @param {number} priority
+   */
+  constructor(callback, priority) {
+    /** @const @private */
+    this.callback_ = callback;
+    /** @const @private */
+    this.priority_ = priority;
+  }
+
+  /**
+   * Calls the callback for transforming/mutating the target object.
+   * @param {!Element} el
+   * @return {!Element}
+   */
+  transform(el) {
+    return this.callback_(el);
```

Comment:
this method might not need to return anything. unless you want to create a new Element instance, which I think is unnecessary and error-prone